src/box/tuple.h

```diff
@@ -1449,6 +1449,18 @@ tuple_to_buf(struct tuple *tuple, char *buf, size_t size);
 size_t
 tuple_runtime_memory_used(void);
 
+/**
+ * Allocate size bytes on runtime arena for tuples.
+ */
+void *
+tuple_runtime_memory_alloc(size_t size);
```

Comment:
I named them likely to `tuple_runtime_memory_used()`

Reply:
I'm not sure we should account these allocations to `tuple_runtime_memory_used()`. After all, we don't account `struct iterator` allocations there, although they are allocated from the same runtime arena. Please discuss this question and the naming with @alyapunov. If we agree that this is in fact tuple memory and we do want to account it in `box.runtime.info().tuple`, then the current names should be fine.

Reply:
- Let's call them `runtime_alloc` and `runtime_free`.
- We don't really want to account these allocations in `box.runtime.info().tuple`, but it is already incorrect because of functional keys, so we decided that it's not a big problem if iterator position will be accounted. Especially, because our iterator position will not live long - 99% of pairs are used with `for` loop.
- We can account memory allocated by `runtime_alloc` in a separate counter and then just subtract it from allocator's used counter. But then we'll want to move there functional keys and so on, and probably add a new stats field. We decided there is no need to make a big deal out of it, and we can just turn a blind eye to a small inaccuracy in the statistics or use malloc to keep it simple.

Reply:
> * We don't really want to account these allocations in `box.runtime.info().tuple`, but it is already incorrect because of functional keys, so we decided that it's not a big problem if iterator position will be accounted. Especially, because our iterator position will not live long - 99% of pairs are used with `for` loop.

FYI functional keys are allocated as memtx tuples so they don't have anything to do with the runtime arena:

https://github.com/tarantool/tarantool/blob/d2d7f4de3a6ba2caa7f260860e750c733ee3201d/src/box/memtx_engine.cc#L1358-L1359